File: backend/src/main/java/com/stresspause/service/TransactionService.java
package com.stresspause.service;

import com.stresspause.dto.request.TransactionRequest;
import com.stresspause.dto.response.TransactionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface TransactionService {
    TransactionResponse createTransaction(TransactionRequest request);

    Page<TransactionResponse> getAllTransactions(Pageable pageable);

    TransactionResponse getTransactionById(UUID id);

    TransactionResponse updateTransaction(UUID id, TransactionRequest request);

    void deleteTransaction(UUID id);
}

File: frontend/src/services/transactionService.js
import api from './api';

const transactionService = {
    fetchTransactions: async (page = 0, size = 10, type = '', category = '') => {
        let url = `/transactions?page=${page}&size=${size}`;
        if (type && type !== 'ALL') url += `&type=${type}`;
        if (category) url += `&category=${category}`;

        const response = await api.get(url);
        return response.data;
    },

    createTransaction: async (data) => {
        const response = await api.post('/transactions', data);
        return response.data;
    },

    updateTransaction: async (id, data) => {
        const response = await api.put(`/transactions/${id}`, data);
        return response.data;
    },

    deleteTransaction: async (id) => {
        const response = await api.delete(`/transactions/${id}`);
        return response.data;
    }
};

export default transactionService;
